src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/RegexNodeConverter.cs

```diff
@@ -418,11 +418,18 @@ BDD Compute(string set)
                     // If the first catCode is negated, the group as a whole is negated
                     bool negatedGroup = categoryCode < 0;
 
-                    // Collect individual category codes
-                    var categoryCodes = new HashSet<UnicodeCategory>();
+                    // Collect individual category codes. At this point, the only codes that can be
+                    // here are valid UnicodeCategories; SpaceConst is never used as part of groups,
+                    // as such groups are only constructed in our own RegexCharClass consts, and it's
+                    // not used there.
+                    const int UnicodeCategoryCount = 1 + (int)UnicodeCategory.OtherNotAssigned;
+                    Debug.Assert(!Enum.IsDefined((UnicodeCategory)UnicodeCategoryCount));
+                    var categoryCodes = new bool[UnicodeCategoryCount];
```

Comment:
Would it make sense to `stackalloc` this (outside of the loop of course) so we aren't allocating an array in a loop?